tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZoneSuite.scala

```diff
@@ -0,0 +1,281 @@
+        // of java.sql.Timestamp, it's convenient to compare result via java.time.Instant
+        .set("spark.sql.datetime.java8API.enabled", "true"))
+
+    // get result from TimeZoneDB
+    val actualRet = withResource(createColumnVector(epochSeconds)) { inputCv =>
+      CpuTimeZoneDB.fromTimestampToUtcTimestamp(
+        inputCv,
+        ZoneId.of(zoneStr))
+    }
+
+    withResource(actualRet) { _ =>
+      assertTimestampRet(actualRet, sparkRet)
+    }
+  }
+
+  def testFromTimestampToDate(epochSeconds: Array[Long], zoneStr: String): Unit = {
+    // get result from Spark
+    val sparkRet = withCpuSparkSession(
+      spark => {
+        createLongDF(spark, epochSeconds).createOrReplaceTempView("tab")
+        // cast timestamp to date
+        // first cast(long as timestamp), it's not timezone awared.
+        spark.sql(s"select cast(cast(ts_long as timestamp) as date) from tab").collect()
+      },
+      new SparkConf()
+        // cast(timestamp as date) will use this timezone
+        .set("spark.sql.session.timeZone", zoneStr)
+        // by setting this, the Spark output for date type is java.time.LocalDate instead
+        // of java.sql.Date, it's convenient to compare result
+        .set("spark.sql.datetime.java8API.enabled", "true"))
+
+    // get result from TimeZoneDB
+    val actualRet = withResource(createColumnVector(epochSeconds)) { inputCv =>
+      CpuTimeZoneDB.fromTimestampToDate(
+        inputCv,
+        ZoneId.of(zoneStr))
+    }
+
+    withResource(actualRet) { _ =>
+      assertDateRet(actualRet, sparkRet)
+    }
+  }
+
+  def testFromDateToTimestamp(epochDays: Array[LocalDate], zoneStr: String): Unit = {
+    // get result from Spark
+    val sparkRet = withCpuSparkSession(
+      spark => {
+        createDateDF(spark, epochDays).createOrReplaceTempView("tab")
+        // cast timestamp to date
+        // first cast(long as timestamp), it's not timezone awared.
+        spark.sql(s"select cast(date_col as Timestamp) from tab").collect()
+      },
+      new SparkConf()
+        // cast(date as timestamp) will use this timezone
+        .set("spark.sql.session.timeZone", zoneStr)
+        // by setting this, the Spark output for date type is java.time.LocalDate instead
+        // of java.sql.Date, it's convenient to compare result
+        .set("spark.sql.datetime.java8API.enabled", "true"))
+
+    // get result from TimeZoneDB
+    val actualRet = withResource(createDateColumnVector(epochDays)) { inputCv =>
+      CpuTimeZoneDB.fromDateToTimestap(
+        inputCv,
+        ZoneId.of(zoneStr))
+    }
+
+    withResource(actualRet) { _ =>
+      assertTimestampRet(actualRet, sparkRet)
+    }
+  }
+
+  test("test all time zones") {
+    assume(false,
+      "It's time consuming for test all time zones, by default it's disabled")
+    //    val zones = ZoneId.getAvailableZoneIds.asScala.toList.map(z => ZoneId.of(z)).filter { z =>
```

Comment:
How about making this as a ```Fuzzer``` and randomly pick up some timezones to verify? We can still reserve tests for basic zone ids (e.g., ```Asia/Shanghai```) below.

Reply:
Now randomly select 2 time zones to test.